src/SuffixArrayConstruction.chpl

```diff
@@ -0,0 +1,220 @@
+module SuffixArrayConstruction {
+  // This module contains algorithms to construct suffix arrays
+
+  //  Chapel implementation of the suffix array construction algorithm using skew algorithm from
+  // "Simple Linear Work Suffix Array Construction" by Juha Karkkainen and Peter Sanders (2003)
+  // Dec.7, 2020
+
+  /*
+  Returns a boolean indicating whether the pair (a1, a2) is less than or equal to pair (b1, b2)
+
+  :arg a1, a2, b1, b2: pairs to be compared
+  :type: int
+
+  :returns: boolean True iff pair (a1, a2) is less than or equal to pair (b1, b2)
+  */
+  inline proc leq_pairs(a1:int, a2:int, b1:int, b2:int): bool {
+    return((a1 < b1) || ((a1 == b1) && (a2 <= b2)));  // lexicographic order for pairs
+  } 
+
+  /*
+  Returns a boolean indicating whether the triple (a1, a2, a3) is less than or equal to triple (b1, b2, b3)
+
+  :arg a1, a2, a3, b1, b2, b3: triples to be compared
+  :type: int
+
+  :returns: boolean True iff triple (a1, a2, a3) is less than or equal to pair (b1, b2, b3)
+  */
+  inline proc leq_triples(a1:int, a2:int, a3:int, b1:int, b2:int, b3:int): bool {
+    return((a1 < b1) || ((a1 == b1) && leq_pairs(a2, a3, b2, b3)));  // lexicographic order for triples
+  }
+
+  /*
+  Using Radix Sort, stably sorts a[0..n-1] according to b[0..n-1] using keys 0..K from r[] 
+  Element a[i] is mapping to r[a[i]] where r is the alphabet with K+1 characters.
+
+  :arg a: array to be sorted
+  :type: [] int
+  :arg b: array used to sort
+  :type: [] int
+  :arg r: array containing keys
+  :type: [] int || uint(8)
+  :arg n: bound for indicies of a and b used during this calculation
+  :type: int
+  :arg K: number of keys
+  :type: int
+  */
+  proc radixPass(a:[] int, b:[] int, r:[] ?t, n:int, K:int) where t == int || t == uint(8) {
+    var c: [0..K] t; // counter array
+    var x: t;
+    var i: int;
+    var tmp: t;
+    var sum=0: int;
+
+    forall x in c do x=0;
+    // calculate the number of different characters in a
+    for i in 0..n-1 do c[r[a[i]]] = c[r[a[i]]]+1;
```

Comment:
I was able to use @ronawho's type declaration trick again to modify  `radixPass` to accept `int` or `uint(8)` eliminating a need for a second function!